### PR TITLE
Improved logging for failed transformation requests

### PR DESCRIPTION
**Pull request by emmberk**
This PR updates `throw new TransformException();` to include the cause. I also added a check to make sure that the fileSize matches the size of the given file.

### CODE REVIEW

**Comment by emmberk** [comment]
I heroed this.

File: ingest/src/test/java/com/connexta/ingest/IngestApplicationIntegrationTest.java [hunk 1]
```diff
@@ -96,11 +98,13 @@ public void testS3UnableToStore() throws Exception {
     when(mockS3Client.putObject(any(PutObjectRequest.class), any(RequestBody.class)))
         .thenThrow(SdkClientException.builder().build());
 
+    final byte[] bytes = "some-content".getBytes();
```
Comment (jhunzik) [inline ingest/src/test/java/com/connexta/ingest/IngestApplicationIntegrationTest.java]: ✏️ Can this be pulled out into the test for the test class?

**Comment by emmberk** [comment]
rebased

**Comment by emmberk** [comment]
build now

**Comment by emmberk** [comment]
```Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)```

build now

**Pull request merged**